src/main.rs

```diff
@@ -155,29 +196,140 @@ fn main(){
+                if output[i].genome == inp.genome{
+                    
+                    for m in inp.methods.iter(){
+                        output[i].methods.push(*m);
+                    }
+                } else {
+                    output.push(inp.clone());
+                }
+                }
+            }
+        return output.clone()
+    }else {
+        let output_len = output.len();
+        let mut output_st = &output.clone()[..input.len()];
+        info!("Lengths Sliced {}, {}", output_st.len(), input.len());
+        for (i, inp) in input.iter().enumerate() {
+            if output_st[i].filename == inp.filename{
+                if output_st[i].genome == inp.genome{
+                    info!("Equal! {:?}{:?}", output[i].genome, inp.genome);
+                    for m in inp.methods.iter(){
+                        output[i].methods.push(*m);
+                    }
+                }
+            // let mut cnt = cnt + 1;
+            }else{
+                info!("Second Check {:?}{:?}", output[output_len-input.len()+i].filename, inp.filename);
+                if output[output_len-input.len()+i].filename == inp.filename{
+                    info!("Third Check {:?}{:?}", output[output_len-input.len()+i].genome, inp.genome);
+                    if output[output_len-input.len()+i].genome == inp.genome{
+                        for m in inp.methods.iter(){
+                            output[output_len-input.len()+i].methods.push(*m);
+                    }
+                    }
+                }
+                else{
+                    output.push(inp.clone())
+                }
+            }
+            }
+        return output.clone()
+    }
+    
+    // else{
+    //     for (i, inp) in input.iter().enumerate() {
+    //         output.push(inp.clone());
+    //     }
+    //     return output.clone()
+    // }
+        
+
+
+    
+    // let mut output_st = Vec::new();
+    // for inp in input{
+    //     output.push(inp);
+    // }
+    // return output.clone()
+}
 
 fn run_genome<R: coverm::bam_generator::NamedBamReader,
               T: coverm::bam_generator::NamedBamReaderGenerator<R>>(
     bam_generators: Vec<T>,
-    m: &clap::ArgMatches) {
+    m: &clap::ArgMatches,
+    output_stream: &mut Vec<OutputStream>,
+    htype: &mut HeaderTypes,
+    method: &str) -> Vec<OutputStream> {
+    
 
-    let method = m.value_of("method").unwrap();
+    let out;
     let min_fraction_covered = value_t!(m.value_of("min-covered-fraction"), f32).unwrap();
+    
     if min_fraction_covered > 1.0 || min_fraction_covered < 0.0 {
         eprintln!("Minimum fraction covered parameter cannot be < 0 or > 1, found {}", min_fraction_covered);
         process::exit(1)
     }
     let print_zeros = !m.is_present("no-zeros");
     let flag_filter = !m.is_present("no-flag-filter");
     let single_genome = m.is_present("single-genome");
-
+    let headers = !m.is_present("remove-headers");
```

Comment:
How about we just keep it simple and always print the headers i.e. remove the remove-headers option? It isn't hard for people to run it through `tail -n+2` if they really don't want headers, yeh?